scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -1290,6 +1290,62 @@
             (catch Exception e
               (log/error e "Unable to decline offers!")))))))
 
+(defn create-mesos-scheduler
+  "Creates the mesos scheduler which offloads processing of status updates to an agent.
+   The agent is used to ensure in-order asynchronous processing of status updates."
```

Comment:
> It is unbounded (limited by heap memory), no back pressure.

This makes me worried. I know we effectively had this with `future` before but can you add a way to monitor how large the queue is getting (have a counter and inc when `async-in-order-processing` and dec when safe call is called) as well as a time spent in queue so we can monitor it?